tasks/kernel_matrix_testing/infra.py

```diff
@@ -126,8 +129,9 @@ def _get_rsync_base(self, exclude: PathOrStr | None, verbose=False) -> str:
             exclude_arg = f"--exclude '{exclude}'"
 
         verbose_arg = "-vP" if verbose else ""
+        sudo = "--rsync-path=\"sudo rsync\"" if self.user != "root" else ""
 
-        return f"rsync {verbose_arg} -e \"ssh {ssh_options_command({'IdentitiesOnly': 'yes'} | SSH_MULTIPLEX_OPTIONS)} {{proxy_cmd}} -i {self.ssh_key}\" -p -rt --exclude='.git*' {exclude_arg} --filter=':- .gitignore'"
+        return f"rsync {sudo} {verbose_arg} -e \"ssh {ssh_options_command({'IdentitiesOnly': 'yes'} | SSH_MULTIPLEX_OPTIONS)} {{proxy_cmd}} -i {self.ssh_key}\" -p -rt --exclude='.git*' {exclude_arg} --filter=':- .gitignore'"
```

Comment:
Same question, can this be `sudo rsync`... instead?

Reply:
That would run rsync as sudo on the host machine. We want `rsync` to have sudo perms on the remote machine so it can write to directories like `/etc/datadog-agent`, etc.